daal4py/sklearn/decomposition/_pca.py

```diff
@@ -0,0 +1,235 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#******************************************************************************/
+
+import numpy as np
+import numbers
+
+from sklearn import decomposition
+from sklearn.utils import check_array
+
+from sklearn.utils.validation import check_is_fitted
+from sklearn.utils.extmath import stable_cumsum
+
+from math import sqrt
+
+import daal4py
+from .._utils import getFPType, get_patch_message, sklearn_check_version
+import logging
+
+if sklearn_check_version('0.22'):
+    from sklearn.decomposition._pca import PCA as PCA_original
+else:
+    from sklearn.decomposition.pca import PCA as PCA_original
+
+if sklearn_check_version('0.23'):
+    from sklearn.decomposition._pca import _infer_dimension
+elif sklearn_check_version('0.22'):
+    from sklearn.decomposition._pca import _infer_dimension_
+else:
+    from sklearn.decomposition.pca import _infer_dimension_
+
+
+class PCA(PCA_original):
+    def __init__(self, n_components=None, copy=True, whiten=False,
+                 svd_solver='auto', tol=0.0, iterated_power='auto',
+                 random_state=None):
+        self.n_components = n_components
+        self.copy = copy
+        self.whiten = whiten
+        self.svd_solver = svd_solver
+        self.tol = tol
+        self.iterated_power = iterated_power
+        self.random_state = random_state
+
+
+    def _validate_n_components(self, n_components, n_samples, n_features):
+        if n_components == 'mle':
+            if n_samples < n_features:
+                raise ValueError("n_components='mle' is only supported "
+                                 "if n_samples >= n_features")
+        elif not 0 <= n_components <= min(n_samples, n_features):
+            raise ValueError("n_components=%r must be between 0 and "
+                             "min(n_samples, n_features)=%r with "
+                             "svd_solver='full'"
+                             % (n_components, min(n_samples, n_features)))
+        elif n_components >= 1:
+            if not isinstance(n_components, numbers.Integral):
+                raise ValueError("n_components=%r must be of type int "
+                                 "when greater than or equal to 1, "
+                                 "was of type=%r"
+                                 % (n_components, type(n_components)))
+
+
+    def _fit_full_daal4py(self, X, n_components):
+        n_samples, n_features = X.shape
+        n_sf_min = min(n_samples, n_features)
+
+        if n_components == 'mle':
+            daal_n_components = n_features
```

Comment:
What is the difference from the number of components from [here](https://github.com/IntelPython/daal4py/pull/383/files#diff-add3ec45396c9436de3ce96cd3c2d9fb5823b3e09c6a798b877cf4efdaf99370R108)?